projects/singularityFinance/index.js

```diff
@@ -0,0 +1,148 @@
+const poolconfig = require('./staking.json');
+
+const sfiToken = {
+  1: "0x7636D8722Fdf7cd34232a915E48e96aA3eB386BF",
+  56: "0x7636D8722Fdf7cd34232a915E48e96aA3eB386BF",
+  8453: "0x7636D8722Fdf7cd34232a915E48e96aA3eB386BF"
+};
+
+const vaultRegistry = {
+  1: [],
+  56: [],
+  8453: ["0x414F0e07cd833cE73c9d59280699f910b48E1ECb"],
+};
+
+const CHAIN_IDS = {
+  ethereum: 1,
+  bsc: 56,
+  base: 8453
+};
+
+// calculates the tvl of the SFI dynavaults by calling totalAsset which is a representation of the vault tvl in in a reference asset
+async function calculateDynaVaultTVL(api, chainId) {
+  for (let i = 0; i < vaultRegistry[chainId].length; i++) {
+    const dynaVaults = await api.multiCall({
+      abi: 'function allVaults() view returns (tuple(address vault, uint8 VaultType, bool active)[] memory)',
+      calls: vaultRegistry[chainId],
+    });
+
+    const assets = await api.multiCall({
+      abi: "function asset() view returns (address)",
+      calls: dynaVaults[0].map(vault => ({
```

Comment:
it seems an odd pattern to iterate over vaultRegistry[chainId].length and then multicall against dynaVaults[0] each time - is this expected? Should we lose the for loop and multicall against each dynaVault?